### PR TITLE
feat: add getText enhancement for complex text widgets

### PR DESCRIPTION
**Pull request by tandt53**
hi there,

Recently I encountered a problem when getting text from Text widget. Then I tried to fix it by revise code in https://github.com/Alpaca00/command-driven-list, and it worked.

- I described my issue and simple solution to fix it here [Solving getText Issues with Appium Flutter Driver: A Complete Guide](https://medium.com/@tandt53/solving-gettext-issues-with-appium-flutter-driver-a-complete-guide-1b2834da5d32).
- I also found a similar issue in [issue#585](https://github.com/appium/appium-flutter-driver/issues/585). 

Thus, I decided to implement a new command `getTextWithCommandExtension` that is similar with `dragAndDropWithCommandExtension`. It works on my local. 

Please refer to my flutter code [here](https://github.com/Alpaca00/command-driven-list/pull/1) and attached video for my demo.

https://github.com/user-attachments/assets/377c004a-7505-4ef2-9774-8599824a8d4d

### CODE REVIEW

File: driver/lib/commands/execute.ts [hunk 1]
```diff
@@ -20,75 +20,78 @@ export const execute = async function(
...
-    case `setFrameSync`:
-      return await setFrameSync(this, args[0], args[1]);
-    case `clickElement`:
-      return await clickElement(this, args[0], args[1]);
-    case `dragAndDropWithCommandExtension`:
-      return await dragAndDropWithCommandExtension(this, args[0]);
-    default:
-      throw new Error(`Command not support: "${rawCommand}"`);
-  }
+    const command = matching[1].trim();
```
Comment (KazuCocoa) [inline driver/lib/commands/execute.ts:23]: Have you applied a formatter? Could you update the indentation? Expected diff would be only `getTextWithCommandExtension` related then

File: driver/lib/commands/execute.ts [hunk 2]
```diff
@@ -242,3 +245,11 @@ const dragAndDropWithCommandExtension = async (
...
   return await self.socket!.executeSocketCommand(commandPayload);
 };
+
+async function getTextWithCommandExtension(self: FlutterDriver, params: { findBy: string; }) {
+    const payload = {
+        command: 'getTextWithCommandExtension',
+        findBy: params.findBy,
+    };
+    return await self.socket!.executeSocketCommand(payload);
+}
```
Comment (KazuCocoa) [inline driver/lib/commands/execute.ts:255]: Could you follow two indentations basis?

File: README.md [hunk 3]
```diff
@@ -374,7 +377,7 @@ void main() {
 }
 ```
 
-#### Simple example using `dragAndDropWithCommandExtension` command in Python
+### Simple example using `dragAndDropWithCommandExtension` command in Python
```
Comment (KazuCocoa) [inline README.md:380]:
```suggestion
#### Simple example using `dragAndDropWithCommandExtension` command in Python
```
?

File: driver/lib/commands/execute.ts (hunk 1 above)
Comment (tandt53) [reply driver/lib/commands/execute.ts]: oops, sorry I set format code automatically in editor, so I was not aware indentation changed. I fixed it.

File: driver/lib/commands/execute.ts (hunk 2 above)
Comment (tandt53) [reply driver/lib/commands/execute.ts]: yep, I fixed it.

File: README.md (hunk 3 above)
Comment (tandt53) [reply README.md]: I fixed it.

File: README.md [hunk 4]
```diff
@@ -395,6 +399,16 @@ payload = {
 driver.execute_script("flutter:dragAndDropWithCommandExtension", payload)
```
[the remark this answers is not in the document]
Comment (tandt53) [reply README.md:398]: I updated python example.

File: README.md [hunk 5]
```diff
@@ -355,21 +355,25 @@ This is a command extension for Flutter Driver, utilizing the [CommandExtension-
...
-Copy the [extended_commands.dart](extended_commands.dart) file to the `lib` folder of your Flutter project.
+Copy the sample dart files to the `lib` folder of your project. Please note that you don't need to copy all files, just copy the file matched with the command you need.
+- dragAndDropWithCommandExtension: [extended_commands.dart](extended_commands.dart)
+- getTextWithCommandExtension: [get_text_command.dart](get_text_command.dart) 
 
 The entry point must include the `List<CommandExtension>?` commands argument in either `main.dart` or `test_main.dart` to properly handle the command extension.
 
 
 ```dart
 import 'extended_commands.dart';
```
[the remark this answers is not in the document]
Comment (tandt53) [reply README.md]: renamed it

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Alpaca00) [reply README.md:398]: Thank you

**Review by KazuCocoa — APPROVED** [review]

**Review by Alpaca00 — APPROVED** [review]

**Comment by KazuCocoa** [comment]
2.11.0 includes this PR.

**Comment by jlipps** [comment]
Hi @tandt53, congrats: the Appium project wants to compensate you for this contribution! Please reply to this comment mentioning @jlipps and @KazuCocoa and sharing your [OpenCollective](https://opencollective.com/) account name, so that we can initiate payment! Or let us know if you decline to receive compensation via OpenCollective. Thank you!

**Comment by tandt53** [comment]
> Hi @tandt53, congrats: the Appium project wants to compensate you for this contribution! Please reply to this comment mentioning @jlipps and @KazuCocoa and sharing your [OpenCollective](https://opencollective.com/) account name, so that we can initiate payment! Or let us know if you decline to receive compensation via OpenCollective. Thank you!

Thank you so much, @jlipps and @KazuCocoa! Contributing to Appium has been such an exciting journey, and I deeply appreciate this incredible gesture. Receiving this compensation as a New Year’s gift would make it even more special. My OpenCollective account name is [tan do](https://opencollective.com/tan-do).

**Pull request merged**